src/Testcontainers.Spanner/SpannerContainer.cs

```diff
@@ -0,0 +1,153 @@
+  {
+    var resource = new
+    {
+      instanceId = _configuration.InstanceId,
+      instance = new
+      {
+        config = "emulator-config",
+        displayName = _configuration.InstanceId,
+      }
+    };
+    string requestUri = $"v1/projects/{_configuration.ProjectId}/instances";
+    string resourceDescription = "instance";
+
+    await CreateResource(requestUri, resource, resourceDescription, ct);
+  }
+
+  private async Task CreateDatabase(CancellationToken ct = default)
+  {
+    var resource = new
+    {
+      createStatement = $"CREATE DATABASE `{_configuration.DatabaseId}`",
+    };
+
+
+    string requestUri = $"v1/projects/{_configuration.ProjectId}/instances/{_configuration.InstanceId}/databases";
+    string resourceDescription = "database";
+
+    await CreateResource(requestUri, resource, resourceDescription, ct);
+  }
+
+  public async Task ExecuteDdlAsync(params string[] ddl)
+  {
+    string requestUri = $"v1/projects/{_configuration.ProjectId}/instances/{_configuration.InstanceId}/databases/{_configuration.DatabaseId}/ddl";
+    var request = new { statements = ddl };
+
+    var response = await _webClient.PatchAsJsonAsync(requestUri, request)
+      .ValidateAsync("update ddl");
+
+    var operation = await response.Content.ReadFromJsonAsync<SpannerOperationDto>() ??
+                    throw new InvalidOperationException(
+                      "Executing ddl resulted in a response with a body deserializing as SpannerOperation to null");
+
+    await AwaitOperationAsync(operation);
+
+    if (operation.Error != null)
+    {
+      throw new InvalidOperationException($"Failed to execute ddl operation with error {operation.Error}");
+    }
+  }
+
+  private async Task AwaitOperationAsync(SpannerOperationDto operation)
+  {
+    int delayInSeconds = 1;
+    const int maxDelayInSeconds = 10;
+
+    while (!operation.Done)
+    {
+      await Task.Delay(TimeSpan.FromSeconds(delayInSeconds));
+      string operationName = operation.Name;
+      var operationResponse = await _webClient.GetAsync($"v1/{operationName}").ValidateAsync("get operation");
+
+      operation = await operationResponse.Content.ReadFromJsonAsync<SpannerOperationDto>() ??
+                  throw new InvalidOperationException(
+                    $"Getting operation resulted in null deserialization of body for operation name {operationName}");
+      if (delayInSeconds * 2 < maxDelayInSeconds)
+      {
+        delayInSeconds *= 2;
+      }
+    }
+  }
+
+
+  private async Task CreateResource(string requestUri, object resource, string resourceDescription, CancellationToken ct = default)
+  {
+    await _webClient.PostAsJsonAsync(requestUri, resource, cancellationToken: ct)
+      .ValidateAsync($"create {resourceDescription}", ct: ct);
+  }
+
+  public void Dispose() => _webClient.Dispose();
+
+  protected override async ValueTask DisposeAsyncCore()
+  {
+    _webClient.Dispose();
+    await base.DisposeAsyncCore();
+  }
```

Comment:
It was inspired by what I saw in the postgresql, where you can include a script for initial setup. In our use cases it made sense to always have this available, but for m ore general purpose i can see it malkes sense to also have the emulator running without this, so I'll remove it from startup.

Reply:
> It was inspired by what I saw in the postgresql, where you can include a script for initial setup.

But this is part (supported) by the PostgreSQL image, right? As previously mentioned, while the intention is nice, we try to avoid incorporating client-side tasks into modules. This is because it does not make sense for us to maintain vendor-specific functionalities when there is already a suitable client that can handle them. Instead, we strive to offer best practice configurations for common test use cases.